pydantic/_internal/_decorators.py

```diff
@@ -192,20 +200,50 @@ class SerializationFunctions(DecoratorFunctions[Serializer]):
 _FUNCS: set[str] = set()
 
 
-def prepare_decorator(function: Callable[..., Any], allow_reuse: bool) -> classmethod[Any]:
+def prepare_validator_decorator(function: Callable[..., Any], allow_reuse: bool) -> Any:
     """
     Convert the function to a classmethod if it isn't already.
 
     Warn about validators/serializers with duplicated names since without this, they can be overwritten silently
     which generally isn't the intended behaviour, don't run in ipython (see #312) or if `allow_reuse` is True.
     """
-    f_cls = function if isinstance(function, classmethod) else classmethod(function)
+    first_param = next(iter(inspect.signature(function).parameters.values()))
+    ret: Any
+    if first_param.name == 'cls':
+        ret = function if isinstance(function, classmethod) else classmethod(function)
+        function = ret.__func__
+    elif first_param.name == 'self':
+        ret = function
+    else:
+        ret = function if isinstance(function, staticmethod) else staticmethod(function)
+        function = ret.__func__
     if not allow_reuse and not in_ipython():
-        ref = f'{f_cls.__func__.__module__}::{f_cls.__func__.__qualname__}'
+        fn = function
+        if isinstance(fn, classmethod):
+            fn = fn.__func__
+        ref = f'{fn.__module__}::{fn.__qualname__}'
         if ref in _FUNCS:
             warnings.warn(f'duplicate validator function "{ref}"; if this is intended, set `allow_reuse=True`')
         _FUNCS.add(ref)
-    return f_cls
+    return ret
+
+
+def prepare_serializer_decorator(function: Callable[..., Any], allow_reuse: bool) -> Any:
+    """
+    Convert the function to a classmethod if it isn't already.
```

Comment:
Does this docstring need changing? It looks like a copy of the above, and I don't see anything happening below that looks like conversion into a classmethod, but maybe I'm missing something.

Maybe this is related to what we talked about with not forcing serializer methods to be classmethods; either way probably makes sense to at least add a TODO in this docstring saying to update it once we have more clarity.